src/solvers/dg/3d/dg.jl

```diff
@@ -1307,11 +1319,120 @@ function rhs!(dg::Dg3D, t_stage)
   @timeit timer() "source terms" calc_sources!(dg, dg.source_terms, t_stage)
 end
 
+# Calculate volume integral and update u_t
+apply_positivity_preserving_limiter!(dg::AbstractDg) = apply_positivity_preserving_limiter!(dg.elements.u, dg, equations(dg))
+
+# Only implemented and tested for compressible Euler equations in 3D
+function apply_positivity_preserving_limiter!(u, dg::AbstractDg, equation::AbstractEquation)
+end
+
+# Apply positivity limiter of Zhan and Shu to nodal values elements.u
+function apply_positivity_preserving_limiter!(u, dg::Dg3D, equation::CompressibleEulerEquations3D)
+  if dg.pp_limiter_apply
+    apply_positivity_preserving_limiter_density!(u, dg::Dg3D, equation::CompressibleEulerEquations3D)
+    apply_positivity_preserving_limiter_pressure!(u, dg::Dg3D, equation::CompressibleEulerEquations3D)
+  end
+end
+
+# Apply positivity limiter of Zhan and Shu to nodal values elements.u
+# density
+function apply_positivity_preserving_limiter_density!(u, dg::Dg3D, equation::CompressibleEulerEquations3D)
+  @unpack weights, pp_limiter_threshold = dg
+  Threads.@threads for element_id in 1:dg.n_elements
+    # Dermine minimum density
+    rho_min = 0.0
```

Comment:
```suggestion
    rho_min = floatmax()
```
Otherwise, `rho_min` will always be zero even if no cell has this value (or is this intentional?)